src/pcv/morpho/MorphoAavePCVDeposit.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {ILens} from "@voltprotocol/pcv/morpho/ILens.sol";
+import {IMorpho} from "@voltprotocol/pcv/morpho/IMorpho.sol";
+import {CoreRefV2} from "@voltprotocol/refs/CoreRefV2.sol";
+import {Constants} from "@voltprotocol/Constants.sol";
+import {PCVDepositV2} from "@voltprotocol/pcv/PCVDepositV2.sol";
+
+/// @notice PCV Deposit for Morpho-Aave V2.
+/// Implements the PCV Deposit interface to deposit and withdraw funds in Morpho
+/// Liquidity profile of Morpho for this deposit is fully liquid for USDC and DAI
+/// because the incentivized rates are higher than the P2P rate.
+/// Only for depositing USDC and DAI. USDT is not in scope.
```

Comment:
you need a separate approval flow where you first 0 token approvals when using USDT, otherwise if you try to set a non zero approval from a non zero approval position, things will revert.